src/pip/_internal/utils/appdirs.py

```diff
@@ -22,255 +20,22 @@
 
 def user_cache_dir(appname):
     # type: (str) -> str
-    r"""
-    Return full path to the user-specific cache dir for this application.
-
-        "appname" is the name of application.
-
-    Typical user cache directories are:
-        macOS:      ~/Library/Caches/<AppName>
-        Unix:       ~/.cache/<AppName> (XDG default)
-        Windows:    C:\Users\<username>\AppData\Local\<AppName>\Cache
-
-    On Windows the only suggestion in the MSDN docs is that local settings go
-    in the `CSIDL_LOCAL_APPDATA` directory. This is identical to the
-    non-roaming app data dir (the default returned by `user_data_dir`). Apps
-    typically put cache data somewhere *under* the given dir here. Some
-    examples:
-        ...\Mozilla\Firefox\Profiles\<ProfileName>\Cache
-        ...\Acme\SuperApp\Cache\1.0
-
-    OPINION: This function appends "Cache" to the `CSIDL_LOCAL_APPDATA` value.
-    """
-    if WINDOWS:
```

Comment:
This one looked suspicious to me as well, but I didn’t really know how to make of the difference. Is the some insight available on how pip elects to set `WINDOWS` this way? When does the edge case occur?

Reply:
The `cli` check was added in 6e1abb91a. So it seems like it’s for IronPython. Some brief Google search tells me we should patch appdirs for it as well.

Reply:
Added a patch in appdirs for this.

Reply:
`WINDOWS` would also apply if `sys.platform.startswith("win")`, though I personally don't know of any case when `sys.platform != "win32" and sys.platform.startswith("win")`.

Reply:
All (currently supported) Windows return `win32`. There was discussion of introducing `win64` IIRC, but CPython elected to continue using `win32` for backwards compatibility. So I guess this assumption is safe until either Microsoft drops 32-bit support entirely, or introduces 128-bit Windows.

Reply:
Since explicit values are given [here](https://docs.python.org/3/library/sys.html#sys.platform), I'm OK with this as-is. Sorry to anyone deviating from those, but this probably isn't the only place they'd have problems!